documentation.md

```diff
@@ -0,0 +1,97 @@
+make sure to modify the top-level `ProtocolTypeRouter` to include the `TurboStreamsConsumer`:
+
+```python
+from turbo.consumers import TurboStreamsConsumer
+application = ProtocolTypeRouter({
+  "http": AsgiHandler(),
+  "websocket": TurboStreamsConsumer.as_asgi()  # Leave off .as_asgi() if using Channels 2.x
+})
+```
+
+#### Including JavaScript
+In our `<head>` for all our templates, we need to `{% include 'turbo/head.html' %}`, which will include the Turbo libraries we need
+in order to use Turbo Drive, Frames and Streams in our app.
+
+### Model Layer
+
+Throughout these docs we'll be using these two basic models for a chat app, similar to the
+[Hotwire demo video](https://www.youtube.com/watch?v=eKY-QES1XQQ):
+
+```python
+# chat/models.py
+
+from django.db import models
+class Room(models.Model):
+    name = models.CharField(max_length=255)
+
+
+class Message(models.Model):
+    room = models.ForeignKey(Room, related_name="messages", on_delete=models.CASCADE)
+    text = models.TextField()
+```
+
+`turbo-django` provides a `BroadcastableMixin` that enables CRUD operations from a given model
+to be broadcast over websockets. To enable messages to be broadcast over websocket to users
+listening in on a particular room, we add the `BroadcastableMixin` and tell `turbo-django`
+that actions on messages should be broadcast to the related `Room` as well:
+
+```python
+class Message(BroadcastableMixin, models.Model):
+    broadcast_to = ["room"]
+    broadcast_self = False
+
+    room = models.ForeignKey(Room, related_name="messages", on_delete=models.CASCADE)
+    text = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+```
+
+`broadcast_to` can contain any foreign key field on the model. We set `broadcast_self` to `False` since we don't
+anticipate listening in to updates on a single message, only on the associated `Room`. This is all the set-up on the
+model side of things that we'll need!
+
+### Template Layer
+
+#### Subscribing to a stream
+Next, we can declare that a given page should subscribe to broadcasts for a given model instance. Make sure to
+`{% load turbo_streams_helpers %}` to get access to the template tag in your template.
+
+Make sure to pass the model instance (here, it's passed as `room`) into the template context. Somewhere in the `<body>`,
+we need to add the template tag `{% turbo_stream_from room %}`. That's all we need to do to connect this template, when
+rendered on the front-end, to the stream of updates for a given model instance!
+
+#### Rendering Streams
+Streams need to render HTML of updates on the backend using template partials to send over to the frontend over the wire.
+`turbo-django` at the moment relies on some conventional names for templates and IDs:
+
+- `app_name/model_name.html` should contain the template partial for rendering the model. The context passed in will
```

Comment:
Maybe this should be overridable with a `broadcast_template_name` class var?

Reply:
Another thought, we could take Django's example with `CreateView` where the template is `{model_name}_form.html` and make the default `{model_name}_stream.html`.

Reply:
I'll work on adding `broadcast_template_name` class var! See comment above about how the stream template is generally supposed to be the same as the template partial used to render the element on the page originally, so that replacing/appending will look uniform. I agree though that there should be some sort of lookup order, so we try out a few different options and don't just assume `app_name/model_name`.

Reply:
I suggest to put that in another issue.

Reply:
Cool! Yes, we can land this PR and iterate.